src/uu/env/src/env.rs

```diff
@@ -323,6 +323,19 @@ impl EnvAppData {
     fn run_env(&mut self, original_args: impl uucore::Args) -> UResult<()> {
         let (original_args, matches) = self.parse_arguments(original_args)?;
 
+        // clap automatically removes '=' from the beginning of the argument, so we need to check for it
+        for arg in &original_args {
+            let arg = arg.to_string_lossy();
+            let prefix = "-u=";
```

Comment:
Wouldn't that break something like `env echo gotcha -u=hello` ?

Reply:
Also, it definitely wouldn't catch `env -vu=hello`.

Suggestion: Instead of trying to parse the arguments again, is there a way to examine `matches` to determine what the exact argument was that leads clap to believe something? This should be much more robust.

Reply:
Nice catch, it really breaks it. Also, I have found some more cases where it breaks things. I am not sure of what approach I should do, because it looks like I am trying to shoehorn too much of if conditions to match the gnu behavior. Some help would be appreciated.

Reply:
> Also, it definitely wouldn't catch `env -vu=hello`.
> 
> Suggestion: Instead of trying to parse the arguments again, is there a way to examine `matches` to determine what the exact argument was that leads clap to believe something? This should be much more robust.

I tried doing something with clap but it turns out that clap removes '=' before the environment variable by itself.